miner/worker.go

```diff
@@ -1039,10 +1042,7 @@ func (env *Work) commitTransactions(mux *event.TypeMux, balanceFee map[common.Ad
 		}
 		go func(logs []*types.Log, tcount int) {
 			if len(logs) > 0 {
-				err := mux.Post(core.PendingLogsEvent{Logs: logs})
-				if err != nil {
-					log.Warn("[commitTransactions] Error when sending PendingLogsEvent", "LogLength", len(logs))
-				}
+				pendingLogsFeed.Send(logs)
 			}
```

Comment:
```go
if len(coalescedLogs) > 0 || env.tcount > 0 {
		// make a copy, the state caches the logs and these logs get "upgraded" from pending to mined
		// logs by filling in the block hash when the block was mined by the local miner. This can
		// cause a race condition if a log was "upgraded" before the PendingLogsEvent is processed.
		cpy := make([]*types.Log, len(coalescedLogs))
		for i, l := range coalescedLogs {
			cpy[i] = new(types.Log)
			*cpy[i] = *l
		}
		go func(logs []*types.Log, tcount int) {
			if len(logs) > 0 {
				pendingLogsFeed.Send(logs)
			}
			if tcount > 0 {
				err := mux.Post(core.PendingStateEvent{})
				if err != nil {
					log.Warn("[commitTransactions] Error when sending PendingStateEvent", "tcount", tcount)
				}
			}
		}(cpy, env.tcount)
	}
```

can be changed to:

```go
if len(coalescedLogs) > 0 {
    ......
}
if env.tcount > 0 {
    ......
}
```